### PR TITLE
All claims fix veteran prefill mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
- Puts `primaryPhone`, `emailAddress`, and `mailingAddress` in the right spots
- Changes a bit of existing logic:
    - Bank info is only prefilled and the 'start in review' property is only set if *all* of the relevant values are available, because of how the bank widget works in relation to all fields being required in the schema
    - Prefill never sends a `formDatea.serviceInformation` property, so it's unnecessary to `_.get()` the existing `serviceInfo` object before adding `servicePeriods` and `reservesNationalGuardService` to it.
- Refactors the prefill transformer to work like the presubmit transformer, which makes it easier to update and maintain.
- Moves the prefill transformer and related helpers into its own file

## Testing done:
- [x] Updates unit tests
- [x] Tested all prefill fields locally using betamocks data

## Screenshots
No change

## Acceptance criteria
- [x] phone and email are prefilled when available
- [x] mailing address prefilled when available
- [x] All other existing prefillable fields work
- [x] tests and test data updated

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 1]
```diff
@@ -17,15 +17,29 @@ import {
...
-  SERVICE_CONNECTION_TYPES,
   USA,
   DATA_PATHS,
   NINE_ELEVEN,
   HOMELESSNESS_TYPES,
   TWENTY_FIVE_MB,
-  disabilityActionTypes,
 } from './constants';
 
+/**
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/utils.jsx]: This function just moved around in `utils`, misleading diff

File: src/applications/disability-benefits/all-claims/tests/prefill-transformer.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    metadata: { test: 'Test Metadata' },
+    formData: { testData: `This isn't getting transformed` },
+    pages: { testPage: 'Page 1' },
+  };
+
+  it('should return a copy of the prefill data', () => {
+    const { pages, formData, metadata } = noTransformData;
+    const transformedPrefill = prefillTransformer(pages, formData, metadata);
+    expect(transformedPrefill).to.not.equal(noTransformData);
+    expect(transformedPrefill).to.deep.equal(noTransformData);
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/prefill-transformer.unit.spec.js]: Heh, this is misleading. :sweat_smile: For a second there, I was like, "Wait, should it equal or should it not? *Make up my mind!*"

**Review by cvalarida — APPROVED** [review]
Looks good! This was deceptively easy to review. I like it.

File: src/applications/disability-benefits/all-claims/prefill-transformer.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    prefillContactInformation,
+    prefillServiceInformation,
+    prefillBankInformation,
+  ];
+
+  const applyTransformations = (data, transformer) => transformer(data);
+
+  return {
+    metadata,
+    formData: transformations.reduce(applyTransformations, form),
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/prefill-transformer.js]: Yeah just formdata. Open to suggestions on the naming here. The problem is one of scoping the `formData` variable.

I have to do: `export default function prefillTransformer(pages, form, metadata)` in order to use the  `formData` name in the transformation functions, e.g.: `const prefillRatedDisabilities = formData => {`

I think what makes more sense though is using something more generic like `data` in the individual transformation functions and using `formData` for the param name in `prefillTransformer`

File: src/applications/disability-benefits/all-claims/tests/prefill-transformer.unit.spec.js (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/prefill-transformer.unit.spec.js]: Yeah, I thought so as well. Probably means I should come up with something better.

**Pull request merged**